Fix edit command bug

limjunxian1 opened the pull request:
This PR addresses bug related issues mentioned in #76. Closes #76

limjunxian1 committed:
Fix edit command bugs

Currently, alternative contact information initially stored is erased
if they are not edited in the edit command.

Also, edit command allows storing the same primary and secondary email.

Fix the bug by changing the hasSameEmail() method and logic flow of edit
command.

limjunxian1 committed:
Fix test case failing issue

limjunxian1 committed:
Merge branch 'master' into implement-addalt

lordidiot commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -207,8 +207,11 @@ public boolean hasValidTelegram() {
      */
-    public boolean hasSameEmail(Email secondaryEmail) {
-        return email.equals(secondaryEmail);
+    public boolean hasSameEmail() {
```
Wondering if `hasRepeatedEmail` would be a more descriptive name?

limjunxian1 replied:
agreed! fixed in 4aff76d

lordidiot commented on src/main/java/seedu/address/logic/commands/EditCommand.java:
```diff
@@ -122,17 +124,34 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
-        } else if (updatedPerson.equals(personToEdit)) {
-            throw new CommandException(String.format(MESSAGE_EDIT_FIELDS_SAME, updatedName));
         } else {
+            if (personToEdit.hasValidBirthday() && updatedBirthday.equals(Optional.empty())) {
```
Wondering if this can be implemented more succinctly. For instance, could we replace the lines above with

```java
Optional<Birthday> updatedBirthday = editPersonDescriptor.getBirthday().or(() -> personToEdit.getBirthday())
```

This way most of these lines below can be removed, other than the last few checks

limjunxian1 replied:
Yes, I absolutely agree! it is my mistake that I am looking at an older Java documentation for the `Optional` class so in a sense, I limited myself to the methods I can use. Thank you for pointing this out and suggesting a great alternative! Fixed in bc09b84

lordidiot reviewed (commented):
Thanks for working on this @limjunxian1, have some comments about a possible cleaner implementation.

limjunxian1 committed:
Improve implementation

Currently, method in EditCommand class is quite long.

Code is shortened by utilizing in built functions from java docs.

limjunxian1 committed:
Rename method name

limjunxian1 committed:
Fix style issues

limjunxian1 commented:
Thank you for the review! I have made the following changes appropriately. Do highlight to me if there is any other issues.

lordidiot reviewed (approved):
Looks good! Thanks for taking the time to squash bugs 👍
